PracticeMode.cs

```diff
@@ -114,6 +116,285 @@ private void HandleSpawnCommand(CCSPlayerController? player, string commandArg,
+				}
+			}
+			else
+			{
+				ReplyToUserCommand(player, $"Usage: .deletenade <name>");
+			}
+	}
+
+
+
+	
+	private void HandleImportNadeCommand(CCSPlayerController? player, string saveNadeName)
+	{
+		if (!isPractice || player == null) return;
+		
+		if (!IsPlayerAdmin(player)) return;
+		
+		if (!string.IsNullOrWhiteSpace(saveNadeName))
+		{
+			
+			// add that into a savedNades list that contains the saveNadeName, playerpos, playerangle
+			string savednadesfileName = "MatchZy/savednades.cfg";
+			string savednadesPath = Path.Join(Server.GameDirectory + "/csgo/cfg", savednadesfileName);
+		
+			if (!File.Exists(savednadesPath)) File.WriteAllLines(savednadesPath, new[] { "Name Location Viewangle" });
+		
+			// Append saveNadeName playerpos playerangle to a new line and save to savednades.cfg
+			var nadeInfo = saveNadeName;
+		
+			File.AppendAllLines(savednadesPath, new[] { nadeInfo });
+		
+			ReplyToUserCommand(player, $" \x0DLineup \x06'{saveNadeName}' \x0Dsaved successfully!");
+			player.PrintToCenter($" \x0DLineup \x06'{saveNadeName}' \x0Dsaved successfully!");
+			ReplyToUserCommand(player, $" \x0DTo load it use \x06.loadnade {saveNadeName}");
+		}
+		else
+		{
+			ReplyToUserCommand(player, $"Usage: .importnade <CODE>");
+		}
+	}
+	
+	private void HandleListNadesCommand(CCSPlayerController? player, string nadeFilter)
+	{
+		if (!isPractice || player == null) return;
+		
+		if (!IsPlayerAdmin(player)) return;
+		
+		// Read the file
+		string savedNadesFileName = "MatchZy/savednades.cfg";
+		string savedNadesPath = Path.Combine(Server.GameDirectory, "csgo", "cfg", savedNadesFileName);
+		
+		if (!string.IsNullOrWhiteSpace(nadeFilter))
+		{
+			ReplyToUserCommand(player, $" \x0D All saved lineups filtered by '{nadeFilter}':");
+		}
+		else
+		{
+			ReplyToUserCommand(player, $" \x0D All saved lineups:");
+		}	
+		if (File.Exists(savedNadesPath))
+		{
+			// Read all lines from the file
+			string[] lines = File.ReadAllLines(savedNadesPath);
+		
+			// Skip the first line
+			for (int i = 1; i < lines.Length; i++)
+			{
+				string line = lines[i];
+			
+				// Split the line into words
+				string[] words = line.Split(' ');
+			
+				if (!string.IsNullOrWhiteSpace(nadeFilter))
+				{
+					// Check if the first word contains the nadeFilter
+					if (words.Length > 0 && words[0].Contains(nadeFilter))
+					{
+						ReplyToUserCommand(player, $" \x0D.loadnade \x06{words[0]}");
+					}
+				}
+				else
+				{
+					// If the filter is empty, just return every first word of the line
+					if (words.Length > 0)
+					{
+						ReplyToUserCommand(player, $"\x0D.loadnade \x06{words[0]}");
+					}
+				}
+			}
+		}
+		else
+		{
+			ReplyToUserCommand(player, "There are no saved nades!.");
+		}
+	}
+
+	
+	private void HandleLoadNadeCommand(CCSPlayerController? player, string loadNadeName)
+	{
+		if (!isPractice || player == null) return;
+		
+		if (!IsPlayerAdmin(player)) return;
```

Comment:
Hey, is there any reason why only admins can use these commands?
I think normal players should also be able to load nades and use other commands, right?

Reply:
right, i suppose saving and loading can be unlocked for non-admin players but ill suggest to leave .deletenade admin only for now so players wont delete a important lineup by accident.

for a future PR i will figure out a way to have non-admin users have their own private lineups as well as the globally saved ones by admins